delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/TokenIssuer.scala

```diff
@@ -0,0 +1,76 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files.routes
+
+import cats.effect.{Clock, IO}
+import com.nimbusds.jose.crypto.{RSASSASigner, RSASSAVerifier}
+import com.nimbusds.jose.jwk.RSAKey
+import com.nimbusds.jose.util.JSONObjectUtils
+import com.nimbusds.jose.{JWSAlgorithm, JWSHeader, JWSObject, Payload}
+import io.circe.{Json, Printer}
+
+import java.security.KeyFactory
+import java.security.interfaces.{RSAPrivateCrtKey, RSAPublicKey}
+import java.security.spec.{PKCS8EncodedKeySpec, RSAPublicKeySpec}
+import java.util.Base64
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
+
+class TokenIssuer(key: RSAKey, tokenValidity: FiniteDuration)(implicit clock: Clock[IO]) {
+  private val signer               = new RSASSASigner(key)
+  private val verifier             = new RSASSAVerifier(key.toPublicJWK)
+  private val TokenValiditySeconds = tokenValidity.toSeconds
+
+  def issueJWSToken(rawPayload: Json): IO[String] =
+    for {
+      now       <- clock.realTimeInstant
+      jwsObject  = mkJWSObject(now.getEpochSecond, rawPayload)
+      _         <- IO.delay(jwsObject.sign(signer))
```

Comment:
We don't need to wrap those in an IO

Reply:
They're Java and say they can throw, it's standard to wrap in IO as an FFI for that case

Reply:
The first one is an IO so they will be in an IO anyway but as you prefer